--- Optimizer.cpp.orig	2010-02-27 13:48:08.000000000 -0600
+++ src/osgUtil/Optimizer.cpp	2010-02-27 13:59:08.000000000 -0600
@@ -1062,12 +1062,11 @@
                 group->setDataVariance(osg::Object::STATIC);
                 group->setNodeMask(transform->getNodeMask());
                 group->setStateSet(transform->getStateSet());
+                group->setUserData(transform->getUserData());
+                group->setDescriptions(transform->getDescriptions());
                 for(unsigned int i=0;i<transform->getNumChildren();++i)
                 {
-                    for(unsigned int j=0;j<transform->getNumParents();++j)
-                    {
-                        group->addChild(transform->getChild(i));
-                    }
+                    group->addChild(transform->getChild(i));
                 }
 
                 for(int i2=transform->getNumParents()-1;i2>=0;--i2)
